osquery/filesystem/tests/filesystem_tests.cpp

```diff
@@ -282,18 +347,42 @@ TEST_F(FilesystemTests, test_no_wild) {
       resolveFilePattern(kFakeDirectory + "/roto.txt", results, GLOB_FILES);
   EXPECT_TRUE(status.ok());
   EXPECT_EQ(results.size(), 1U);
-  EXPECT_TRUE(contains(results, kFakeDirectory + "/roto.txt"));
+  EXPECT_TRUE(contains(
+      results,
+      fs::path(kFakeDirectory + "/roto.txt").make_preferred().string()));
 }
 
 TEST_F(FilesystemTests, test_safe_permissions) {
   // For testing we can request a different directory path.
+#ifdef WIN32
```

Comment:
Instead of mixing in several `ifdef`s, can this set the several platform-related directory variables once, then call the same APIs outside of the platform-specific code?